test/e2e_node/mirror_pod_test.go

```diff
@@ -196,6 +198,51 @@ var _ = SIGDescribe("MirrorPod", func() {
 			}, 2*time.Minute, time.Second*4).Should(gomega.BeNil())
 		})
 	})
+
+	ginkgo.Context("static pod admit failed when kubelet start", func() {
+		ginkgo.It("should become Failed and have reason UnexpectedAdmissionError [NodeConformance] [Serial]", func() {
+			ns := f.Namespace.Name
+			podPath := framework.TestContext.KubeletConfig.StaticPodPath
+
+			staticPodName := "static-pod" + string(uuid.NewUUID())
+			mirrorPodName := staticPodName + "-" + framework.TestContext.NodeName
+
+			ginkgo.By("create the static pod")
+			err := createStaticPod(podPath, staticPodName, ns,
+				imageutils.GetE2EImage(imageutils.Nginx), v1.RestartPolicyAlways)
+			framework.ExpectNoError(err)
+
+			ginkgo.By("wait for the mirror pod to be running")
+			gomega.Eventually(func() error {
+				return checkMirrorPodRunning(f.ClientSet, mirrorPodName, ns)
+			}, 2*time.Minute, time.Second*4).Should(gomega.BeNil())
+
+			// modify maxPods to 1 to trigger admission failure for static pod
+			oldCfg, err := getCurrentKubeletConfig()
+			framework.ExpectNoError(err)
```

Comment:
This should move into a `BeforeEach` and use https://github.com/kubernetes/kubernetes/blob/4569e646ef161c0262d433aed324fec97a525572/test/e2e_node/util.go#L162 to update the config to ensure it's rolled back

Reply:
Done, I need the case that mirror pod has already exists, so I should modify kubelet config after static pod's creation